Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -252,6 +254,33 @@ namespace xr
             std::vector<Frame::Plane> Planes{};
             std::vector<FeaturePoint> FeaturePointCloud{};
         } ActionResources{};
+            
+        struct HandJointData {
+            void InitializeXRHandJoints()
+            {
+                locations.jointCount = XR_HAND_JOINT_COUNT_EXT;
+                locations.jointLocations = jointLocations;
+            };
+
+            XrHandJointLocationEXT jointLocations[XR_HAND_JOINT_COUNT_EXT];
+            XrHandJointLocationsEXT locations{XR_TYPE_HAND_JOINT_LOCATIONS_EXT};
+        };
+
+        struct HandInfo {
+            XrHandEXT hand{};
+            XrHandTrackerEXT handTracker{};
+            XrSpace handSpace{};
+
+            HandJointData handJointData{};
+        };
+        
+        struct
+        {
+            XrBool32 supportsHandTracking{ false };
+            XrBool32 HandsInitialized{ false };
+
+            std::array<HandInfo, 2> handsInfo;
+        } HandData;
```

Comment:
>HandData [](start = 10, length = 8)

nit: would anyone ever want to access this hand data in a custom extension? Are there different hand related openxr extensions that may never map to webxr? It may be worth considering whether to add this to the XrRegistry/XrSessionContext compared to having it inaccessible to other components here.

If there are no known scenarios where this information would be needed outside of the hand tracking implementation in this component, theres no need to move this.

Reply:
It sounds like there are some HL2-specific pieces at the moment (e.g. pointing/arm ray) that aren't part of the WebXR spec and that aren't currently known to be supported on other devices (e.g. Quest). Because of that we may want to consider making whatever pieces necessary to light that up published outside of this module.